jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/BuildImageMojo.java

```diff
@@ -113,45 +115,47 @@ public void execute() throws MojoExecutionException, MojoFailureException {
       optionalToCredential =
           pluginConfigurationProcessor
               .getMavenSettingsServerCredentials()
-              .retrieve(targetImage.getRegistry());
+              .retrieve(targetImageReference.getRegistry());
       optionalToCredential.ifPresent(
           toCredential ->
               defaultCredentialRetrievers.setInferredCredential(
                   toCredential, MavenSettingsServerCredentials.CREDENTIAL_SOURCE));
     }
     defaultCredentialRetrievers.setCredentialHelperSuffix(getTargetImageCredentialHelperName());
 
-    ImageConfiguration targetImageConfiguration =
-        ImageConfiguration.builder(targetImage)
-            .setCredentialRetrievers(defaultCredentialRetrievers.asList())
-            .build();
+    RegistryImage targetImage = RegistryImage.named(targetImageReference);
+    defaultCredentialRetrievers.asList().forEach(targetImage::addCredentialRetriever);
 
     try {
-      BuildConfiguration buildConfiguration =
+      JibContainerBuilder jibContainerBuilder =
           pluginConfigurationProcessor
-              .getBuildConfigurationBuilder()
-              .setBaseImageConfiguration(
-                  pluginConfigurationProcessor.getBaseImageConfigurationBuilder().build())
-              .setTargetImageConfiguration(targetImageConfiguration)
-              .setAdditionalTargetImageTags(getTargetImageAdditionalTags())
-              .setContainerConfiguration(
-                  pluginConfigurationProcessor.getContainerConfigurationBuilder().build())
-              .setTargetFormat(ImageFormat.valueOf(getFormat()).getManifestTemplateClass())
-              .build();
+              .getJibContainerBuilder()
+              // Only uses possibly non-Docker formats for build to registry.
+              .setFormat(ImageFormat.valueOf(getFormat()));
+
+      Containerizer containerizer = Containerizer.to(targetImage);
+      PluginConfigurationProcessor.configureContainerizer(
+          containerizer, this, mavenProjectProperties);
 
       HelpfulSuggestions helpfulSuggestions =
           new MavenHelpfulSuggestionsBuilder(HELPFUL_SUGGESTIONS_PREFIX, this)
-              .setBaseImageReference(buildConfiguration.getBaseImageConfiguration().getImage())
+              .setBaseImageReference(pluginConfigurationProcessor.getBaseImageReference())
               .setBaseImageHasConfiguredCredentials(
                   pluginConfigurationProcessor.isBaseImageCredentialPresent())
-              .setTargetImageReference(buildConfiguration.getTargetImageConfiguration().getImage())
+              .setTargetImageReference(targetImageReference)
               .setTargetImageHasConfiguredCredentials(optionalToCredential.isPresent())
               .build();
```

Comment:
We can move everything up to here out of the try block since they don't throw the exceptions.